sycl/include/CL/sycl/detail/pi.h

```diff
@@ -159,6 +160,15 @@ const pi_uint64 PI_MEM_FLAGS_ACCESS_RW     = CL_MEM_READ_WRITE;
 const pi_uint64 PI_MEM_FLAGS_HOST_PTR_USE  = CL_MEM_USE_HOST_PTR;
 const pi_uint64 PI_MEM_FLAGS_HOST_PTR_COPY = CL_MEM_COPY_HOST_PTR;
 
+// NOTE: queue properties is implemented this way to better support bit
+// manipulations
+typedef pi_uint64 pi_queue_properties;
+const pi_uint64 PI_QUEUE_OUT_OF_ORDER_EXEC_MODE_ENABLE =
```

Comment:
would it be worth using "const pi_queue_properties ..." rather than "const pi_uint64 ..." here? And if so would it be worth replicating that to pi_mem_flags? I imagine it would be harder to accidentally break something if the typedef for pi_mem_flag or pi_queue_properties was ever to change.

Reply:
Yep, good idea.